opensipscli/modules/trap.py

```diff
@@ -41,7 +41,18 @@ def get_pids(self):
         except:
             self.pids = []
 
-    def get_gdb_output(self, process, pid):
+    def get_gdb_output(self, pid):
+        if os.path.islink("/proc/{}/exe".format(pid)):
+            # get process line of pid
+            process = os.readlink("/proc/{}/exe".format(pid))
+        else:
+            logger.error("could not find OpenSIPS process {} running on local machine".format(pid))
+            return -1
+        # Check if process is opensips (can be different if CLI is running on another host)
+        path, filename = os.path.split(process)
+        if filename != PROCESS_NAME:
+            logger.error("process ID {} is not OpenSIPS process".format(pid))
+            return -1
```

Comment:
This is a significant improvement to the current code.  One could argue that there still may be issues when running multiple OpenSIPS on the same box, with the `trap` command potentially debugging a different instance than the intended one (i.e. the _current one_, see the `-i` argument of `opensips-cli`), but let's cross that river when we get there.